runtime/binding-http/src/main/java/io/aklivity/zilla/runtime/binding/http/internal/stream/HttpServerFactory.java

```diff
@@ -6252,8 +6255,14 @@ private void validatePseudoHeaders(
                         scheme++;
                         break;
                     default:
-                        streamError = Http2ErrorCode.PROTOCOL_ERROR;
-                        return;
+                        if (headers.containsKey(HEADER_NAME_METHOD)
+                            && METHOD_NAME_CONNECT.equals(headers.get(HEADER_NAME_METHOD))
+                            && HEADER_NAME_PROTOCOL.equals(name.getStringWithoutLengthUtf8(0, name.capacity()))
+                            && headers.get(HEADER_NAME_PROTOCOL).length() > 0) {
+                            break;
+                        } else {
+                            streamError = Http2ErrorCode.PROTOCOL_ERROR;
+                        }
```

Comment:
Here we are just tracking the presence and count of the pseudo-headers, so they can be checked in `decodeHeaders`. So we likely need to add a count for `protocol` and include in the check.

Note we are validating the pseudo-headers themselves, not the values.

Reply:
I think I misread this part of the spec in `rfc8441`:

_The :protocol pseudo-header field MUST be included in the CONNECT request, and it MUST have a value of websocket to initiate a WebSocket connection on an HTTP/2 stream_

I now think it means that if we have a CONNECT request, before we create a websocket connection we must ensure that the request also contains the `:protocol` pseudo-header and it must have the value `websocket`.

Maybe we don't need to do any validation in the http2 binding in relation to a connect request.

Reply:
Until we find a better place for it, I have added a check to allow a `:protocol` pseudo-header in a CONNECT request which is otherwise not allowed by `rfc7540`. This seems to make sense to be in the http2 binding because the binding is always advertising the `ENABLE_CONNECT_PROTOCOL` setting.

Reply:
Agreed, just make sure you are doing the simple count here, and the enforcement in `decodeHeaders`, aligned with approach for other checks.